bulk/formatter.go

```diff
@@ -0,0 +1,87 @@
+package bulk
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// Record is the interface to the fields of the bulk uploader record.
+type Record interface {
+	Fields() map[string]interface{}
+	InsertNull() bool
+}
+
+// Formatter is the object that will add records for the bulk uploader.
+type Formatter struct {
+	job    *Job
+	fields []string
+	sb     strings.Builder
+}
+
+// NewFormatter creates a new formatter using the job and the list of fields.
+func NewFormatter(job *Job, fields []string) (*Formatter, error) {
+	if job == nil {
+		return nil, errors.New("bulk formatter: job is required for the formatter")
+	}
+	if len(fields) == 0 {
+		return nil, errors.New("bulk formatter: fields are required")
+	}
+
+	f := &Formatter{
+		job:    job,
+		fields: fields,
+		sb:     strings.Builder{},
+	}
+	_, err := f.sb.WriteString(strings.Join(fields, job.delimiter()))
```

Comment:
I _think_ this line could get combined with the `if` check if you prefer.

Reply:
thanks for the idea